src/messaging/ReliableMessageManager.cpp

```diff
@@ -36,31 +36,66 @@
 namespace chip {
 namespace Messaging {
 
-ReliableMessageManager::RetransTableEntry::RetransTableEntry() :
-    rc(nullptr), msgBuf(nullptr), msgId(0), msgSendFlags(0), nextRetransTimeTick(0), sendCount(0)
-{}
+ReliableMessageManager::RetransTableEntry::RetransTableEntry() : rc(nullptr), peerNodeId(0), nextRetransTimeTick(0), sendCount(0) {}
 
-ReliableMessageManager::ReliableMessageManager() :
-    mTimeStampBase(System::Timer::GetCurrentEpoch()), mCurrentTimerExpiry(0),
+ReliableMessageManager::ReliableMessageManager(std::array<ExchangeContext, CHIP_CONFIG_MAX_EXCHANGE_CONTEXTS> & contextPool) :
+    mContextPool(contextPool), mSystemLayer(nullptr), mCurrentTimerExpiry(0),
     mTimerIntervalShift(CHIP_CONFIG_RMP_TIMER_DEFAULT_PERIOD_SHIFT)
 {}
 
 ReliableMessageManager::~ReliableMessageManager() {}
 
-void ReliableMessageManager::ProcessDelayedDeliveryMessage(ReliableMessageContext * rc, uint32_t PauseTimeMillis)
+void ReliableMessageManager::Init(chip::System::Layer * systemLayer, SecureSessionMgr * sessionMgr)
+{
+    mSystemLayer = systemLayer;
+    mSessionMgr  = sessionMgr;
+
+    memset(mRetransTable, 0, sizeof(mRetransTable));
```

Comment:
The app can call init function multiple times, this guarantee the retransTable is in a clear state after ReliableMessageManager is initialized. 

ReliableMessageManager has the same life-cycle as ExchangeMgr, systemLayer and sessionMgr are not valid when its constructor is called.

Reply:
> The app can call init function multiple times

That is ... not how functions called "Init" usually behave.  If the intent is that this is allowed then (1) that should be explicitly documented and (2) just doing the memset like this will leak whatever buffers are in the table, so it's not the right way to clear the table.

Or is the intent that you have to call `Shutdown` between two calls to `Init`?  If so, that will clear the table, and then the `memset` here won't leak, but of course at that point `Shutdown` has zeroed things out, so the `memset` is not needed either.

> systemLayer and sessionMgr are not valid when its constructor is called.

I see, ok.  Worth documenting on the API.